### PR TITLE
Hw7 networking

### PR DESCRIPTION
**Pull request by tohachug**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -51,4 +51,8 @@ dependencies {
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
     implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:2.2.0"
     implementation "androidx.lifecycle:lifecycle-livedata-ktx:2.2.0"
+    implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
+    implementation 'com.jakewharton.retrofit:retrofit2-kotlinx-serialization-converter:0.8.0'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.0'
```
Comment (Axrorxoja) [inline app/build.gradle:57]: okhttp not need 
this dependency has inside retrofit dependency

File: app/src/main/java/com/example/myapplicationwork/util/ResProvider.kt [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
-import android.content.Context
+import com.example.myapplicationwork.GetInfoFromApi
 import com.example.myapplicationwork.data.Movie
-import com.example.myapplicationwork.data.loadMovies
 
-class ResProvider(val context: Context): Provider {
+class ResProvider() : Provider {
     override suspend fun loadFilms(): List<Movie> {
-        return loadMovies(context)
+        val getInfoFromApi = GetInfoFromApi()
```
Comment (Axrorxoja) [inline app/src/main/java/com/example/myapplicationwork/util/ResProvider.kt]: let's `getInfoFromApi` variable as global
and create it via lazy or inside init block

**Comment by tohachug** [comment]
'com.squareup.okhttp3:okhttp:4.9.0' оставил, без него не работает ""application/json".toMediaType()"

**Comment by Axrorxoja** [comment]
> 'com.squareup.okhttp3:okhttp:4.9.0' оставил, без него не работает ""application/json".toMediaType()"

okay

**Review by Axrorxoja — APPROVED** [review]

**Pull request merged**